python/hyperspace/hyperspace.py

```diff
@@ -103,6 +103,16 @@ def vacuumIndex(self, indexName):
         """
         self.hyperspace.vacuumIndex(indexName)
 
+    def vacuumOutdatedDataIndex(self, indexName):
```

Comment:
nit: for conciseness and consistency, a shorter verb would be nice, such as "clean" (or "clean up"), "prune", "trim", or "shrink".

Reply:
As we discussed, using a flag in vacuumIndex
```scala
vacuumIndex(indexName: String, exceptLatest: Boolean = False)
```
or I think `shrinkIndex` looks better to me

Reply:
Should I use shrinkIndex?

Reply:
IMHO, putting this feature in the same API function as "vacuumIndex", is like putting "deleteIndex" and "vacuumIndex" in the same function. They perform similar actions, but different enough to be called by different names.

Regarding the word choice, I'm not a native speaker, but it seems "clean (up)" or "prune" is more appropriate here for the feature than "shrink". Maybe we'd better consult a native English speaker on this matter.

Reply:
> IMHO, putting this feature in the same API function as "vacuumIndex", is like putting "deleteIndex" and "vacuumIndex" in the same function. They perform similar actions, but different enough to be called by different names.

cc @rapoth who suggested using `vacuumIndex` with a flag.